app/views/articles/_sticky_nav.html.erb

```diff
@@ -55,16 +55,4 @@
   <% end %>
 </div>
 
-<%# cache("article-sidebar-content-#{rand(5)}-#{@article.id}-#{user_signed_in?}-#{(@organization || @user).latest_article_updated_at}", expires_in: 15.minutes) do %>
-  <% sidebar_ad = DisplayAd.for_display(area: "post_sidebar",
-                                        user_signed_in: user_signed_in?,
-                                        organization_id: @article.organization_id,
-                                        article_tags: @article.decorate.cached_tag_list_array,
-                                        article_id: @article.id,
-                                        permit_adjacent_sponsors: @article.decorate.permit_adjacent_sponsors?) %>
-  <% if sidebar_ad %>
-    <div class="crayons-article-sticky grid gap-4 break-word pt-4">
-      <%= render partial: "shared/display_ad", locals: { display_ad: sidebar_ad, data_context_type: DisplayAdEvent::CONTEXT_TYPE_ARTICLE } %>
-    </div>
-  <% end %>
-<%# end %>
+<div class="crayons-article-sticky grid gap-4 break-word pt-4 js-display-ad-container" data-async-url="<%= "/#{@article.username}/#{@article.slug}/display_ads/post_sidebar" %>"></div>
```

Comment:
It happens that I've been reviewing #19312 recently and it's got me thinking a lot about how we ought to be preserving locale information in routes. What about this?

```suggestion
<div class="crayons-article-sticky grid gap-4 break-word pt-4 js-display-ad-container" data-async-url="<%= article_display_ad_path placement_area: :post_sidebar %>"></div>
```

(`:username`, `:slug`, and even potentially `:locale` will autofill from params this way).